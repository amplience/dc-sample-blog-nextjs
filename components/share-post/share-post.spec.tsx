import React from 'react';
import renderer from 'react-test-renderer';
import SharePost from './share-post';

describe('SharePost', (): void => {
  it('should render using the process.env.URL is defined', (): void => {
    process.env.BASE_URL = 'http://example.com/';
    const twitterText = 'Testing Title';
    const wrapper = renderer.create(<SharePost twitterText={twitterText} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when process.env.BASE_URL is undefined', (): void => {
    delete process.env.BASE_URL;
    const twitterText = 'Testing Title';
    const wrapper = renderer.create(<SharePost twitterText={twitterText} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
