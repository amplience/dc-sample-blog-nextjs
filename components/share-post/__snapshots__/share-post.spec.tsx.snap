// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharePost should render using the process.env.URL is defined 1`] = `
<section
  className="jsx-2442235184"
>
  <h3
    className="jsx-2442235184"
  >
    Share the post
  </h3>
  <div
    className="jsx-2442235184"
  >
    <div
      className="jsx-2442235184"
    >
      <script
        className="jsx-2442235184"
        data-url="mocked-path"
        type="IN/Share"
      />
    </div>
    <div
      className="jsx-2442235184"
    >
      <a
        className="jsx-2442235184 twitter-share-button"
        data-show-count="false"
        data-text="Testing Title"
        href="https://twitter.com/share?ref_src=twsrc%5Etfw"
      >
        Tweet
      </a>
    </div>
  </div>
</section>
`;

exports[`SharePost should render when process.env.BASE_URL is undefined 1`] = `
<section
  className="jsx-2442235184"
>
  <h3
    className="jsx-2442235184"
  >
    Share the post
  </h3>
  <div
    className="jsx-2442235184"
  >
    <div
      className="jsx-2442235184"
    >
      <script
        className="jsx-2442235184"
        data-url="mocked-path"
        type="IN/Share"
      />
    </div>
    <div
      className="jsx-2442235184"
    >
      <a
        className="jsx-2442235184 twitter-share-button"
        data-show-count="false"
        data-text="Testing Title"
        href="https://twitter.com/share?ref_src=twsrc%5Etfw"
      >
        Tweet
      </a>
    </div>
  </div>
</section>
`;
